### PR TITLE
Update Skill Description to be more intuitive

### LINKED ISSUE #691: Reformat descriptions
image
image
Note that italics should only be on the dark gray parts*

### LINKED ISSUE #782: Skills that use energy over time use the value on click
i.e. blizzard uses energy over time, when first using it, it will take the energy value (from get energy), and then drain that value at a rate of value/second.

### LINKED ISSUE #921: Pestilence config/description needs an update
config does not match other skills

### PR DESCRIPTION
**Pull request by StudentAlleg**
Update descriptions to be more descriptive and easier to use.

- [x] Add an easy method to display numerical values
- [x] Add tags for what skills do (i.e. damage, movement)
- [x] Add descriptions for effects in skills
- [x] Update Assassin
- [x] Update Mage
- [x] Update Ranger
- [x] Update Brute
- [x] Update Knight
- [x] Update Warlock

Fixes #691
Fixes #782
Fixes #921

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have tested my changes.

### CODE REVIEW

**Comment by StudentAlleg** [comment]
@Mykindos Just want to double check that this is the right path, before I go through and implement this for all skills.

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/knight/passives/LevelField.java [hunk 1]
```diff
@@ -29,7 +30,7 @@
 
 @Singleton
 @BPvPListener
-public class LevelField extends Skill implements Listener {
+public class LevelField extends Skill implements Listener, DefensiveSkill {
```
Comment (Mykindos) [inline champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/knight/passives/LevelField.java:33]: Also offfensive / damage skill

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/knight/sword/Riposte.java [hunk 2]
```diff
@@ -41,7 +42,7 @@
 
 @Singleton
 @BPvPListener
-public class Riposte extends ChannelSkill implements CooldownSkill, InteractSkill {
+public class Riposte extends ChannelSkill implements CooldownSkill, InteractSkill, OffensiveSkill, DamageSkill, HealthSkill {
```
Comment (Mykindos) [inline champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/knight/sword/Riposte.java:45]: also defensive

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/warlock/axe/Cleanse.java [hunk 3]
```diff
@@ -24,7 +24,7 @@
 
 @Singleton
 @BPvPListener
-public class Cleanse extends Skill implements InteractSkill, CooldownSkill, Listener {
+public class Cleanse extends Skill implements InteractSkill, CooldownSkill, Listener, DefensiveSkill {
```
Comment (Mykindos) [inline champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/warlock/axe/Cleanse.java:27]: Team skill too

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/warlock/axe/TormentedSoil.java [hunk 4]
```diff
@@ -35,7 +37,7 @@
 
 @Singleton
 @BPvPListener
-public class TormentedSoil extends Skill implements InteractSkill, CooldownSkill, Listener {
+public class TormentedSoil extends Skill implements InteractSkill, CooldownSkill, Listener, HealthSkill, AreaOfEffectSkill, OffensiveSkill {
```
Comment (Mykindos) [inline champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/warlock/axe/TormentedSoil.java:40]: Also technically a team skill, everyone who stands in it gets bonus damage

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/warlock/passives/SoulHarvest.java [hunk 5]
```diff
@@ -26,7 +27,7 @@
 
 @Singleton
 @BPvPListener
-public class SoulHarvest extends Skill implements PassiveSkill {
+public class SoulHarvest extends Skill implements PassiveSkill, BuffSkill {
```
Comment (Mykindos) [inline champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/warlock/passives/SoulHarvest.java:30]: heal skill, if regeneration qualifies

File: core/src/main/java/me/mykindos/betterpvp/core/energy/EnergyHandler.java [hunk 6]
```diff
@@ -14,6 +15,7 @@
 import org.bukkit.entity.Player;
 
 @Singleton
+@Slf4j
```
Comment (Mykindos) [inline core/src/main/java/me/mykindos/betterpvp/core/energy/EnergyHandler.java:18]: remove

**Review by Mykindos — CHANGES_REQUESTED** [review]

File: core/src/main/java/me/mykindos/betterpvp/core/energy/EnergyHandler.java (hunk 6 above)
Comment (StudentAlleg) [reply core/src/main/java/me/mykindos/betterpvp/core/energy/EnergyHandler.java:18]:
```suggestion
```

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/warlock/passives/SoulHarvest.java (hunk 5 above)
Comment (StudentAlleg) [reply champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/warlock/passives/SoulHarvest.java:30]: I don't think I did this with other regeneration skills, just left it as a buff skill. Though is a gray area.

**Review by Mykindos — APPROVED** [review]

**Pull request merged**